File: app/src/main/java/me/taste2plate/app/customer/repo/ReviewRepository.java
package me.taste2plate.app.customer.repo;


import me.taste2plate.app.customer.common.WooLiveData;
import me.taste2plate.app.Woocommerce;
import me.taste2plate.app.models.ProductReview;
import me.taste2plate.app.models.filters.ProductReviewFilter;

import javax.inject.Inject;
import java.util.List;

public class ReviewRepository {

    @Inject
    Woocommerce woocommerce;

    @Inject
    public ReviewRepository() {
    }


    public WooLiveData<List<ProductReview>> reviews(int productId) {
        final WooLiveData<List<ProductReview>> callBack = new WooLiveData();

        ProductReviewFilter filter = new ProductReviewFilter();
        int[] products = {productId};

        filter.setProduct(products);

        woocommerce.ReviewRepository().reviews(filter).enqueue(callBack);
        return callBack;
    }

    public WooLiveData<ProductReview> create(ProductReview review) {
        final WooLiveData<ProductReview> callBack = new WooLiveData();
        woocommerce.ReviewRepository().create(review).enqueue(callBack);
        return callBack;
    }


}

File: woodroid/src/main/java/me/taste2plate/app/repo/product/ReviewRepository.kt
package me.taste2plate.app.repo.product

import me.taste2plate.app.data.api.ProductReviewAPI
import me.taste2plate.app.models.ProductReview
import me.taste2plate.app.models.filters.ProductReviewFilter
import me.taste2plate.app.repo.WooRepository
import retrofit2.Call

class ReviewRepository(baseUrl: String, consumerKey: String, consumerSecret: String) :
    WooRepository(baseUrl, consumerKey, consumerSecret) {

    private val apiService: ProductReviewAPI

    init {
        apiService = retrofit.create(ProductReviewAPI::class.java)
    }

    fun create(review: ProductReview): Call<ProductReview> {
        return apiService.create(review)
    }


    fun review(id: Int): Call<ProductReview> {
        return apiService.view(id)
    }

    fun reviews(): Call<List<ProductReview>> {
        return apiService.list()
    }

    fun reviews(productReviewFilter: ProductReviewFilter): Call<List<ProductReview>> {
        return apiService.filter(productReviewFilter.filters)
    }

    fun update(id: Int, review: ProductReview): Call<ProductReview> {
        return apiService.update(id, review)
    }

    fun delete(id: Int): Call<ProductReview> {
        return apiService.delete(id)
    }

    fun delete(id: Int, force: Boolean): Call<ProductReview> {
        return apiService.delete(id, force)
    }


}
